src/xhyve.c

```diff
@@ -773,6 +776,57 @@ firmware_parse(const char *opt) {
 	return -1;
 }
 
+static void
+remove_pidfile()
+{
+	int error;
+
+	if (pidfile == NULL)
+		return;
+
+	error = unlink(pidfile);
+	if (error < 0)
+		fprintf(stderr, "Failed to remove pidfile\n");
+}
+
+static int
+setup_pidfile()
+{
+	int f, error, pid;
+	char pid_str[21];
+
+	if (pidfile == NULL)
+		return 0;
+
+	pid = getpid();
+
+	error = sprintf(pid_str, "%d", pid);
+	if (error < 0)
+		return -1;
```

Comment:
I don't think that all these arbitrary (and possibly overlapping with elsewhere) exit codes are a good idea.  It's not clear what each ones value means.  I suggest writing out the error to stderr and using EXIT_FAILURE or possibly using sysexits.h if you want more verbosity.